In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df = pd.read_excel(
    r"C:\Users\Suresh\Downloads\sample_-_superstore.xls",
    engine="xlrd"
)

In [3]:
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,CA-2016-152156,2016-11-08,2016-11-11,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136
1,2,CA-2016-152156,2016-11-08,2016-11-11,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820
2,3,CA-2016-138688,2016-06-12,2016-06-16,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714
3,4,US-2015-108966,2015-10-11,2015-10-18,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310
4,5,US-2015-108966,2015-10-11,2015-10-18,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164


In [4]:
df.shape

(9994, 21)

In [5]:
df.columns

Index(['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode',
       'Customer ID', 'Customer Name', 'Segment', 'Country', 'City', 'State',
       'Postal Code', 'Region', 'Product ID', 'Category', 'Sub-Category',
       'Product Name', 'Sales', 'Quantity', 'Discount', 'Profit'],
      dtype='object')

In [6]:
df.isnull().sum()

Row ID           0
Order ID         0
Order Date       0
Ship Date        0
Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Postal Code      0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
Quantity         0
Discount         0
Profit           0
dtype: int64

In [7]:
df.duplicated().sum()

np.int64(0)

# Data Understanding

The dataset contains sales transactions from a retail store.

It contains information about:
- Orders
- Customers
- Products
- Sales
- Profit
- Discounts
- Regions

The dataset has 9994 rows and 21 columns.

No major missing values were found.

# Exploratory Data Analysis (EDA)

## Business Question 1:
Which category generates the highest sales?

In [9]:
df.groupby("Category")["Sales"].sum()

Category
Furniture          741999.7953
Office Supplies    719047.0320
Technology         836154.0330
Name: Sales, dtype: float64

### Observation

Technology generated the highest sales with total sales of 836,154.03.

Furniture generated sales of 741,999.80, while Office Supplies generated sales of 719,047.03.

Therefore, Technology is the best-performing category in terms of revenue.

## Business Question 2:
Which region generates the highest sales?

In [10]:
df.groupby("Region")["Sales"].sum()

Region
Central    501239.8908
East       678781.2400
South      391721.9050
West       725457.8245
Name: Sales, dtype: float64

### Observation

The West region generated the highest sales with total sales of 725,457.82.

The East region generated sales of 678,781.24, followed by the Central region with 501,239.89.

The South region recorded the lowest sales at 391,721.91.

Therefore, the West region is the best-performing region in terms of revenue generation.

## Business Question 3:
Which category generates the highest profit?

In [11]:
df.groupby("Category")["Profit"].sum()

Category
Furniture           18451.2728
Office Supplies    122490.8008
Technology         145454.9481
Name: Profit, dtype: float64

### Observation

Technology generated the highest profit with a total profit of 145,454.95.

Office Supplies generated a profit of 122,490.80.

Furniture generated the lowest profit at 18,451.27.

Although Furniture achieved high sales, its profit was significantly lower compared to Technology and Office Supplies.

Therefore, Technology is the most profitable category in the dataset.

## Business Question 4:
Do discounts affect profit?

In [12]:
df[["Discount", "Profit"]].corr()

,Discount,Profit
Discount,1.000000,-0.219487
Profit,-0.219487,1.000000


### Observation

The correlation between Discount and Profit is -0.219.

This indicates a negative relationship between discounts and profit.

As discounts increase, profit tends to decrease.

Therefore, the hypothesis that higher discounts reduce profit is supported by the data.

## Business Question 5:
Which sub-categories generate the highest sales?

In [13]:
df.groupby("Sub-Category")["Sales"].sum() \
  .sort_values(ascending=False) \
  .head(10)

Sub-Category
Phones         330007.0540
Chairs         328449.1030
Storage        223843.6080
Tables         206965.5320
Binders        203412.7330
Machines       189238.6310
Accessories    167380.3180
Copiers        149528.0300
Bookcases      114879.9963
Appliances     107532.1610
Name: Sales, dtype: float64

### Observation

Phones generated the highest sales with total sales of 330,007.05.

Chairs closely followed with sales of 328,449.10.

Storage, Tables, and Binders were also among the top-performing sub-categories.

The top 5 revenue-generating sub-categories are:
1. Phones
2. Chairs
3. Storage
4. Tables
5. Binders

This indicates that these product groups contribute significantly to the company's overall revenue.

## Business Question 6:
Which orders generated the highest losses?

In [15]:
df.nsmallest(10, "Profit")[["Product Name", "Category", "Sales", "Discount", "Profit"]]

,Product Name,Category,Sales,Discount,Profit
7772,Cubify CubeX 3D Printer Double Head Print,Technology,4499.985,0.7,-6599.9780
683,Cubify CubeX 3D Printer Triple Head Print,Technology,7999.980,0.5,-3839.9904
9774,GBC DocuBind P400 Electric Binding System,Office Supplies,2177.584,0.8,-3701.8928
3011,Lexmark MX611dhe Monochrome Laser Printer,Technology,2549.985,0.7,-3399.9800
4991,Ibico EPK-21 Electric Binding System,Office Supplies,1889.990,0.8,-2929.4845
3151,Cubify CubeX 3D Printer Double Head Print,Technology,1799.994,0.7,-2639.9912
5310,Fellowes PB500 Electric Punch Plastic Comb Bin...,Office Supplies,1525.188,0.8,-2287.7820
9639,Chromcraft Bull-Nose Wood Oval Conference Tabl...,Furniture,4297.644,0.4,-1862.3124
1199,GBC DocuBind P400 Electric Binding System,Office Supplies,1088.792,0.8,-1850.9464
2697,Cisco TelePresence System EX90 Videoconferenci...,Technology,22638.480,0.5,-1811.0784


### Observation

The analysis identified several orders that generated significant losses.

The largest loss was caused by the product "Cubify CubeX 3D Printer Double Head Print", which resulted in a loss of 6,599.98.

Most loss-making products belong to the Technology and Office Supplies categories.

A common pattern among these products is the presence of very high discounts (50% to 80%).

This suggests that excessive discounting has a strong negative impact on profitability and should be carefully managed.

Therefore, the company's discount strategy should be reviewed to reduce financial losses and improve overall profit margins.

## Business Question 7:
Which orders generated the highest profits?

In [16]:
df.nlargest(10, "Profit")[["Product Name", "Category", "Sales", "Discount", "Profit"]]

,Product Name,Category,Sales,Discount,Profit
6826,Canon imageCLASS 2200 Advanced Copier,Technology,17499.950,0.0,8399.9760
8153,Canon imageCLASS 2200 Advanced Copier,Technology,13999.960,0.0,6719.9808
4190,Canon imageCLASS 2200 Advanced Copier,Technology,10499.970,0.0,5039.9856
9039,GBC Ibimaster 500 Manual ProClick Binding System,Office Supplies,9892.740,0.0,4946.3700
4098,Ibico EPK-21 Electric Binding System,Office Supplies,9449.950,0.0,4630.4755
2623,Canon imageCLASS 2200 Advanced Copier,Technology,11199.968,0.2,3919.9888
509,Fellowes PB500 Electric Punch Plastic Comb Bin...,Office Supplies,6354.950,0.0,3177.4750
8488,HP Designjet T520 Inkjet Large Format Printer ...,Technology,8749.950,0.0,2799.9840
7666,Hewlett Packard LaserJet 3310 Copier,Technology,5399.910,0.0,2591.9568
6520,GBC DocuBind P400 Electric Binding System,Office Supplies,5443.960,0.0,2504.2216


### Observation

The product "Canon imageCLASS 2200 Advanced Copier" generated the highest profit of 8,399.98.

Most of the highest-profit products belong to the Technology category, indicating strong profitability in this segment.

Several high-profit products were sold with little or no discount, suggesting that maintaining healthy pricing contributes significantly to profitability.

Office Supplies also contributed several highly profitable products, particularly binding systems and office equipment.

These products should be prioritized in marketing, inventory planning, and sales strategies to maximize revenue and profit.

# EDA Summary

1. The dataset contains 9,994 sales records and 21 variables related to orders, customers, products, sales, profit, and discounts.

2. No major missing values were found in the dataset.

3. Technology generated the highest sales (836,154.03) among all categories.

4. The West region generated the highest sales (725,457.82), making it the best-performing region.

5. Technology was also the most profitable category with a profit of 145,454.95.

6. Phones and Chairs were the highest-selling sub-categories.

7. The correlation between Discount and Profit was -0.219, indicating that higher discounts tend to reduce profit.

8. Several products generated substantial losses due to high discounts ranging from 50% to 80%.

9. The Canon imageCLASS 2200 Advanced Copier was the most profitable product in the dataset.

10. The analysis suggests that discount management and focusing on high-performing technology products can improve overall profitability.

# Conclusion

This exploratory data analysis was conducted on the Superstore sales dataset to understand sales performance, profitability, regional trends, and discount impacts.

The analysis revealed that Technology is the strongest category in terms of both sales and profit. The West region contributed the highest revenue, while Phones and Chairs emerged as the leading sub-categories by sales.

Statistical analysis showed a negative relationship between discounts and profit, indicating that excessive discounting can significantly reduce profitability. Additionally, anomaly detection identified several loss-making products associated with high discount levels.

Overall, the findings provide valuable business insights that can help improve sales strategies, pricing decisions, and inventory management.